generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpec.java

```diff
@@ -141,7 +141,10 @@ public FieldSpec withoutType(IsOfTypeConstraint.Types type){
         return new FieldSpec(null, restrictions.put(type, restriction), nullable, formatting);
     }
 
-    public boolean isTypeAllowed(IsOfTypeConstraint.Types type){
+    public boolean allowsRestrictionForType(IsOfTypeConstraint.Types type){
```

Comment:
I think I prefer the previous function name, assuming it's doing effectively the same thing.